market_maker_keeper/uniswapv2_market_maker_keeper.py

```diff
@@ -0,0 +1,441 @@
+                            help="Ethereum account from which to send transactions")
+
+        parser.add_argument("--eth-key", type=str, nargs='*',
+                            help="Ethereum private key(s) to use (e.g. 'key_file=aaa.json,pass_file=aaa.pass')")
+
+        parser.add_argument("--pair", type=str, required=True,
+                            help="Token pair (sell/buy) on which the keeper will operate")
+
+        parser.add_argument("--token-config", type=str, required=True,
+                            help="Token configuration file")
+
+        parser.add_argument("--uniswap-feed", type=str, required=True,
+                            help="Source of liquidity feed")
+
+        parser.add_argument("--uniswap-feed-expiry", type=int, default=86400,
+                            help="Maximum age of the liquidity feed (in seconds, default: 86400)")
+
+        parser.add_argument("--gas-price", type=int, default=9000000000,
+                            help="Gas price (in Wei)")
+
+        parser.add_argument("--smart-gas-price", dest='smart_gas_price', action='store_true',
+                            help="Use smart gas pricing strategy, based on the ethgasstation.info feed")
+
+        parser.add_argument("--initial-exchange-rate", type=float, default=None,
+                            help="Used to determine the initial ratio to be used in a newly created pool")
+
+        parser.add_argument("--percentage-difference", type=float, default=2,
+                            help="Percentage difference between Uniswap exchange rate and aggregated price"
+                                 "(default: 2)")
+
+        parser.add_argument("--uniswap-percentage-difference", type=float, default=5,
+                            help="Percentage difference between future Uniswap exchange rate and aggregated price"
+                                 "(default: 5)")
+
+        parser.add_argument("--debug", dest='debug', action='store_true',
+                            help="Enable debug output")
+
+        self.arguments = parser.parse_args(args)
+        setup_logging(self.arguments)
+
+        self.web3 = kwargs['web3'] if 'web3' in kwargs else Web3(HTTPProvider(endpoint_uri=f"{self.arguments.rpc_host}:{self.arguments.rpc_port}",
+                                                                              request_kwargs={"timeout": self.arguments.rpc_timeout}))
+
+        self.web3.eth.defaultAccount = self.arguments.eth_from
+        register_keys(self.web3, self.arguments.eth_key)
+
+        # Record if eth is in pair, so can check which liquidity method needs to be used
+        self.is_eth = 'ETH' in self.pair()
+
+        # Assume token b is always ETH if ETH is in pair
+        token_a_name = self.pair().split('-')[0]
+        token_b_name = 'WETH' if self.pair().split('-')[1] == 'ETH' or self.pair().split('-')[1] == 'WETH' else self.pair().split('-')[1]
+
+        self.reloadable_config = ReloadableConfig(self.arguments.token_config)
```

Comment:
I don't think token config should be passed as an argument but since the config json is bundled within keeper code it should read directly from that file.

Reply:
That makes sense, I will adjust accordingly.

Reply:
Yes, the goal here is to provide generalized support for `Token` class. This config file can then be reused across different keepers, which would have let us simplify DyDx as just one example.